lib/iron_bank/actions/query.rb

```diff
@@ -6,10 +6,25 @@ module Actions
     # https://knowledgecenter.zuora.com/DC_Developers/K_Zuora_Object_Query_Language
     #
     class Query < Action
+      def self.call(zoql, limit: 0)
+        new(zoql, limit).call
+      end
+
       private
 
+      attr_reader :zoql, :limit
+
+      def initialize(zoql, limit)
+        @zoql  = zoql
+        @limit = limit
+      end
+
       def params
-        { 'queryString': args }
+        if limit.zero?
```

Comment:
maybe use merge to default params on `unless limit.zero?`

Reply:
[personal opinion] Having the two params in an `if/else` block reads better than:

```rb
def params
  params = { queryString: zoql }
  params.merge(conf: { batchSize: limit }) unless limit.zero?

  params
end
```

or

```rb
def params
  { queryString: zoql }.tap do |params|
    params.merge(conf: { batchSize: limit }) unless limit.zero?
  end
end
```

I _think_ the following reads better, but I'm open if you have another suggestion or vote for one or the other form:

```rb
def params
  if limit.zero?
    { queryString: zoql }
  else
    { queryString: zoql, conf: { batchSize: limit } }
  end
end
```

Reply:
Ended up doing:

```rb
def params
  return required_params if limit.zero?

  required_params.merge(conf: { batchSize: limit })
end

def required_params
  { queryString: zoql }
end
```

Good suggestion, thanks @mustafaturan!